packages/x-charts/src/hooks/useAxisEvents.ts

```diff
@@ -121,7 +129,24 @@ export const useAxisEvents = (disableAxisListener: boolean) => {
       const newStateX = getNewAxisState(xAxis[usedXAxis], svgPoint.x);
       const newStateY = getNewAxisState(yAxis[usedYAxis], svgPoint.y);
 
-      dispatch({ type: 'updateAxis', data: { x: newStateX, y: newStateY } });
+      store.update((prev) => ({
+        ...prev,
+        interaction: {
+          ...prev.interaction,
+          axis: {
+            // A bit verbose, but prevent losing the x value if only y got modified.
+            ...prev.interaction.axis,
+            ...(prev.interaction.axis.x?.index !== newStateX?.index ||
+            prev.interaction.axis.x?.value !== newStateX?.value
+              ? { x: newStateX }
+              : {}),
+            ...(prev.interaction.axis.y?.index !== newStateY?.index ||
+            prev.interaction.axis.y?.value !== newStateY?.value
+              ? { y: newStateY }
+              : {}),
+          },
+        },
+      }));
```

Comment:
What about using `structuredClone` to make it more readable? 

```suggestion
      store.update((prev) => {
        const newState = structuredClone(prev);
        
        if (
          prev.interaction.axis.x?.index !== newStateX?.index ||
          prev.interaction.axis.x?.value !== newStateX?.value
        ) {
          newState.interaction.axis.x = newStateX;
        }

        if (
          prev.interaction.axis.y?.index !== newStateY?.index ||
          prev.interaction.axis.y?.value !== newStateY?.value
        ) {
          newState.interaction.axis.y = newStateY;
        }

        return newState;
      });
```

Reply:
Not sure there are perf impact though

Reply:
I tested, but it does not work. As you can see moving the mouse along the y-axis create a re-render of the tooltip (I added a console log in the tooltip component)

https://github.com/user-attachments/assets/7382468d-a2ae-42a7-9d49-a133005f21e6

Reply:
:(

I suppose it is because of object equality 🤔